cpp/src/strings/regex/regex.cuh

```diff
@@ -35,17 +34,15 @@ class string_view;
 namespace strings {
 namespace detail {
 
-struct reljunk;
-struct reinst;
-class reprog;
+struct relist;
 
 using match_pair   = thrust::pair<cudf::size_type, cudf::size_type>;
 using match_result = thrust::optional<match_pair>;
 
-constexpr int32_t RX_STACK_SMALL  = 112;    ///< fastest stack size
-constexpr int32_t RX_STACK_MEDIUM = 1104;   ///< faster stack size
-constexpr int32_t RX_STACK_LARGE  = 10128;  ///< fast stack size
-constexpr int32_t RX_STACK_ANY    = 8;      ///< slowest: uses global memory
+constexpr int32_t RX_STACK_SMALL  = 112;   ///< fastest stack size
+constexpr int32_t RX_STACK_MEDIUM = 1104;  ///< faster stack size
+constexpr int32_t RX_STACK_LARGE  = 2560;  ///< fast stack size
```

Comment:
Can we document how we got to these magic values. Why is 2560 preferred over 10128?

Reply:
The comment just below explains how these are based on the instruction counts.
Honestly, these are all going away in the next PR.

Reply:
Okay, I found the comment. But now the comment is out of date. Is 2560 now equal to 1000 instructions or has the number of instructions gone down?

Reply:
The comment is still correct for 10128 and 1000.
The 2560 is 252 instructions: `int(2560*10.125)`